packages/SwingSet/src/kernel/vat-loader/vat-loader.js

```diff
@@ -219,6 +95,12 @@ export function makeVatLoader(stuff) {
       name,
     } = options;
 
+    if (isDynamic && enableDisavow) {
+      throw Error(`dynamic vat ${vatID} using forbidden enableDisavow`);
+    }
+    if (!isDynamic && meterID !== undefined) {
+      throw Error(`static vat ${vatID} given meterID ${meterID}, no`);
+    }
```

Comment:
Yeah, it's disallowing specific combinations. This function receives `RecordedVatOptions`, which is no longer distinguished into static and dynamic types. These prohibitions ought to be have been enforced earlier, when we converted the static/dynamic inputs into the common recorded form, so these checks are a backup.

I'll update the error messages as recommended. I'm going to stick with the separate prohibitions, though, rather than the `else` branch, or `Fail`, or `q()`.. I think it's more readable this way:

```js
    if (isDynamic && enableDisavow) {
      throw Error(`dynamic vat ${vatID} must not use option enableDisavow`);
    }
    if (!isDynamic && meterID !== undefined) {
      throw Error(`static vat ${vatID} must not use option meterID`);
    }
```